### ISSUE TITLE
Question from character level RNN classifier, why not use the hidden state across epochs?

### ISSUE THREAD

**Issue by labJunky** [spro/practical-pytorch#139]
In the RNN classification [example](https://github.com/spro/practical-pytorch/blob/master/char-rnn-classification/char-rnn-classification.ipynb), using characters of names to predict the names language, the train function re-zeros the hidden state (and gradient) every epoch. I was wondering why this is done, instead of carrying over the final hidden states of the epoch before?

**Comment by ZhouXing19** [comment]
One epoch means a run-through of a word. If we start a new epoch, which means we are training the network with a new word, we need to redefine the hidden state of the initial letter of the new word, since states of different words are independent.
